qutip/models/dicke.py

```diff
@@ -0,0 +1,1325 @@
+    Dicke basis. If the argument `basis` is "uncoupled" then it generates
+    the state in a 2**N dim Hilbert space.
+    """
+    if basis == "uncoupled":
+        return _uncoupled_excited(N)
+
+    jmm1 = {(N/2, N/2, N/2): 1}
+    return dicke_basis(N, jmm1)
+
+def superradiant(N, basis = "dicke"):
+    """
+    Generates the superradiant dicke state as |N/2, 0> or |N/2, 0.5>
+    in the Dicke basis
+    """
+    if basis == "uncoupled":
+        return _uncoupled_superradiant(N)
+
+    if N%2 == 0:
+        jmm1 = {(N/2, 0, 0):1.}
+        return dicke(N, jmm1)
+    else:
+        jmm1 = {(N/2, 0.5, 0.5):1.}
+    return dicke_basis(N, jmm1)
+
+def css(N, a=1/np.sqrt(2), b=1/np.sqrt(2), basis = "dicke"):
+    """
+    Loads the separable spin state |CSS>= Prod_i^N(a|1>_i + b|0>_i)
+    into the reduced density matrix rho(j,m,m'). 
+    The default state is the symmetric CSS, |CSS> = |+>.
+    """
+    if basis == "uncoupled":
+        return _uncoupled_css(N, a, b)
+
+    nds = _num_dicke_states(N)
+    num_ladders = _num_dicke_ladders(N)
+    rho = dok_matrix((nds, nds))
+
+    # loop in the allowed matrix elements
+    jmm1_dict = jmm1_dictionary(N)[1]
+
+    j = 0.5 * N 
+    mmax = int(2 * j + 1)
+    for i in range(0, mmax):
+        m = j - i
+        psi_m = np.sqrt(float(energy_degeneracy(N, m))) * a**( N * 0.5 + m) * b**( N * 0.5 - m)
+        for i1 in range(0, mmax):
+            m1 = j - i1
+            row_column = jmm1_dict[(j, m, m1)]
+            psi_m1 = np.sqrt(float(energy_degeneracy(N, m1))) * a**( N * 0.5 + m1) * b**( N * 0.5 - m1)
+            rho[row_column] = psi_m * psi_m1
+    
+    return Qobj(rho)
+
+def ghz(N, basis = "dicke"):
+    """
+    Generates the the density matric of the GHZ state
+    """
+    if basis == "uncoupled":
+        return _uncoupled_ghz(N)
+
+    nds = _num_dicke_states(N)
+    rho = dok_matrix((nds, nds))
+
+    rho[0,0] = 1/2
+    rho[N,N] = 1/2
+    rho[N,0] = 1/2
+    rho[0,N] = 1/2
+
+    return Qobj(rho)
+
+def ground(N, basis = "dicke"):
+    """
+    Generates the density matric of the ground state for N spins
+    """
+    if basis == "uncoupled":
+        return _uncoupled_ground(N)
+
+    nds = _num_dicke_states(N)
+    rho = dok_matrix((nds, nds))
+
+    rho[-1, -1] = 1
+
+    return Qobj(rho)
+
+def uncoupled_identity(N):
+    """
+    Generates the identity in a 2**N dimensional Hilbert space
+
+    Parameters
+    ----------
+    N: int
+        The number of two level systems
+
+    Returns
+    -------
+    identity: Qobj matrix (QuTiP class)
+        With the correct dimensions (dims)
+    """
+    N = int(N)
+    rho = np.zeros((2**N, 2**N))
+    for i in range(0, 2**N):
+        rho[i, i] = 1
+
+    spin_dim = [2 for i in range(0, N)]
+    spins_dims = list((spin_dim, spin_dim))
+    identity = Qobj(rho, dims = spins_dims)
+
+    return identity
+
+# Uncoupled states in the full Hilbert space.
```

Comment:
Some of these 'uncoupled' functions are defined as internal others external. It's not clear to me why